src/ol/layer/MapboxVector.js

```diff
@@ -405,10 +405,7 @@ class MapboxVectorLayer extends VectorTileLayer {
     }
 
     const source = this.getSource();
-    if (
-      styleSource.url.indexOf('mapbox://') === 0 ||
-      styleSource.url.indexOf('{z}') !== -1
```

Comment:
I removed this because according to the spec, a `url` in the source can never contain a tile url template.